react/features/toolbar/components/Toolbar.web.js

```diff
@@ -0,0 +1,409 @@
+
+    /**
+     * Unregisters listeners before unmounting the component.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        this._unregisterListeners();
+    }
+
+    /**
+     * Unregisters listeners of some UI events.
+     *
+     * @private
+     * @returns {void}
+     */
+    _unregisterListeners() {
+        APP.UI.removeListener(UIEvents.SIDE_TOOLBAR_CONTAINER_TOGGLED,
+            this._onSideToolbarContainerToggled);
+
+        APP.UI.removeListener(UIEvents.LOCAL_RAISE_HAND_CHANGED,
+            this._onLocalRaiseHandChanged);
+
+        APP.UI.removeListener(UIEvents.FULLSCREEN_TOGGLED,
+            this._onFullScreenToggled);
+
+        APP.UI.removeListener(UIEvents.SHOW_CUSTOM_TOOLBAR_BUTTON_POPUP,
+            this._onShowCustomToolbarPopup);
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <div>
+                {
+                    this._renderPrimaryToolbar()
+                }
+                {
+                    this._renderSecondaryToolbar()
+                }
+            </div>
+        );
+    }
+
+    /**
+     * Renders primary toolbar.
+     *
+     * @returns {ReactElement}
+     * @private
+     */
+    _renderPrimaryToolbar() {
+        const renderPrimaryToolbarButton = (acc, button, index) => {
+            const isSplitter = interfaceConfig.MAIN_TOOLBAR_SPLITTER_INDEX
+                && index === interfaceConfig.MAIN_TOOLBAR_SPLITTER_INDEX;
+
+            if (isSplitter) {
+                const splitter = <span className = 'toolbar__splitter' />;
+
+                acc.push(splitter);
+            }
+
+            const toolbarButton = this._renderToolbarButton(button);
+
+            acc.push(toolbarButton);
+
+            return acc;
+        };
+
+        return (
+            <div id = 'mainToolbarContainer'>
+                <div
+                    className = 'notice'
+                    id = 'notice'
+                    style = { _DISPLAY_NONE_STYLE }>
+                    <span
+                        className = 'noticeText'
+                        id = 'noticeText' />
+                </div>
+                <div
+                    className = 'toolbar'
+                    id = 'mainToolbar'>
+                    {
+                        [ ...this.state.primaryToolbar.values() ]
+                            .reduce(renderPrimaryToolbarButton, [])
+                    }
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Renders toolbar button.
+     *
+     * @param {Object} button - Object representing the button.
+     * @returns {ReactElement}
+     * @private
+     */
+    _renderToolbarButton(button) {
+        const onClick = event => {
+            const handler = BUTTON_HANDLERS[button.id];
+
+            if (!$(event.target).prop('disabled') && handler) {
```

Comment:
Intention of this PR was just to start generate toolbar and toolbar buttons via React. I've got rid of jquery in the code based on this branch.